server/gokbg3/src/main/groovy/org/gokb/UpdatePkgTippsRun.groovy

```diff
@@ -0,0 +1,730 @@
+            errors : [message: ge.toString()]
+        ]
+        tipp?.expunge()
+        return tipp_error
+      }
+      if (tipp) {
+        if (existing_tipp_ids.size() > 0 && existing_tipp_ids.contains(tipp.id)) {
+          log.debug("Existing TIPP matched!")
+          existing_tipp_ids.removeElement(tipp.id)
+        }
+        // Probably, these tipp.status are overwritten already
+        if (tipp.status != status_deleted && tippJson.status == "Deleted") {
+          tipp.deleteSoft()
+          removedNum++;
+        }
+        else if (tipp.status != status_retired && tippJson.status == "Retired") {
+          tipp.retire()
+          removedNum++;
+        }
+        else if (tipp.status != status_current && (!tippJson.status || tippJson.status == "Current")) {
+          if (tipp.isDeleted() && !fullsync) {
+            reviewRequestService.raise(
+                tipp,
+                "Matched TIPP was marked as Deleted.",
+                "Check TIPP Status.",
+                user,
+                null,
+                null,
+                rr_deleted
+            )
+          }
+          tipp.status = status_current
+        }
+        tipp.merge()
+        if (!tipp.hostPlatform) {
+          log.debug("unknown hostPlatform for TIPP $tipp")
+        }
+        else if (tipp.isCurrent() && tipp.hostPlatform.status != status_current) {
+          def additionalInfo = [:]
+          additionalInfo.vars = [tipp.hostPlatform.name, tipp.hostPlatform.status?.value]
+          reviewRequestService.raise(
+              tipp,
+              "The existing platform matched for this TIPP (${tipp.hostPlatform}) is marked as ${tipp.hostPlatform.status?.value}! Please review the URL/Platform for validity.",
+              "Platform not marked as current.",
+              user,
+              null,
+              (additionalInfo as JSON).toString(),
+              rr_nonCurrent
+          )
+        }
+      }
+      else {
+        log.debug("Could not reference TIPP")
+        invalidTipps << tippJson
+        def tipp_error = [
+            message: messageService.resolveCode('crossRef.package.tipps.error', [tippJson.title.name], locale),
+            baddata: tippJson
+        ]
+        return tipp_error
+      }
+    }
+    return tippError
+  }
+
+/**
+ * this method finds similar tipps based on their identifiers (ids). for performance reasons, the ElasticSearch index
+ * is used first and if it fails, a database search is performed after for a definitive decision.
+ * @param tippJson
+ * @return
+ */
+  private TitleInstancePackagePlatform[] findTipps(tippJson) {
+    def tipps = []
+    // search TIPPs for json.title_id == tipp.importId
+    if (tippJson.titleId) {
+      // elastic search
+      TypeConvertingMap map = [
+          componentType    : 'TitleInstancePackagePlatform',
+          importId         : tippJson.titleId,
+          tippPackageUuid  : pkg.uuid,
+          hostPlatformUuid : tippJson.hostPlatform.uuid,
```

Comment:
Use `pkg` & `platform` as query params for `find()`, as documented at https://github.com/openlibraryenvironment/gokb/wiki/Elasticsearch-API (component-specific fields).